### PR TITLE
fix: use realpaths for symlink source and target

### PR DESCRIPTION
**Pull request by vjpr**
https://github.com/pnpm/pnpm/issues/1208

### CODE REVIEW

File: packages/supi/src/link.ts [hunk 1]
```diff
@@ -209,13 +210,17 @@ async function linkToModules (
     prefix: string,
   },
 ) {
-  const dest = path.join(opts.destModulesDir, opts.alias)
-  const {reused} = await symlinkDir(opts.packageDir, dest)
+
+  const destModulesDirReal = fs.realpathSync(opts.destModulesDir)
+  const packageDirReal = fs.realpathSync(opts.packageDir)
```
Comment (zkochan) [inline packages/supi/src/link.ts]: use `import fs = require('mz/fs')`

and the async version `await fs.realpath(...`

**Comment by zkochan** [comment]
Failed with error:

```
ERROR: /home/travis/build/pnpm/pnpm/packages/supi/src/link.ts[39, 1]: Import sources within a group must be alphabetized.
```

**Comment by zkochan** [comment]
also, commit format should follow these guidelines: https://github.com/pnpm/pnpm/blob/master/CONTRIBUTING.md#commit-message-format

**Comment by vjpr** [comment]
@zkochan PR merges are squashed right? So you don't need me to rewrite the commits? I will use correct format going forward though.

**Comment by zkochan** [comment]
ok

**Comment by zkochan** [comment]
seems like this change is breaking a few tests though

**Comment by vjpr** [comment]
11 test failures in supi. Will investigate.

**Comment by vjpr** [comment]
The issue is `opts.destModulesDir` (some `node_modules` dir) sometimes doesn't exist in the tests, which then throws an error.

```
const destModulesDirReal = fs.realpathSync(opts.destModulesDir)
```

However, `symlinkDir` works when the dir doesn't exist.

**Comment by zkochan** [comment]
Wouldn't the solution be, if the dir does not exist, just pass it to `symlink-dir`. If it does not exist yet then no need to get the real location of it

**Comment by vjpr** [comment]
> If it does not exist yet then no need to get the real location of it

But a parent dir may be a symlink, so we need to get the realpath of the parent dir. I'm not sure if this will happen.

I'm doing this:

```
  let destModulesDirReal
  try {
    destModulesDirReal = await fs.realpath(opts.destModulesDir)
  } catch (err) {
    if (err.code === 'ENOENT') {
      await mkdirp(opts.destModulesDir)
      destModulesDirReal = await fs.realpath(opts.destModulesDir)
    } else {
      throw err
    }
  }
```

**Comment by vjpr** [comment]
I've added a test. I added a fixture of the workspace from my original issue to the `test/packages` dir.

**Comment by vjpr** [comment]
@zkochan Looks like AppVeyor failed during network issue. How do we restart the check?

**Comment by vjpr** [comment]
@zkochan Maybe something to do with symlinks.

I added symlinks inside the fixture - I'm guessing they don't work cross platform.

I have to programmatically create the symlinks I assume.

**Comment by zkochan** [comment]
I see, that might be the issues indeed

**Pull request merged**